src/cli.py

```diff
@@ -42,479 +104,1158 @@ def _format_action_invocation(self, action):
-            default=default, required=required, **kwargs
+            option_strings, dest, nargs=nargs, const=const, default=default, 
+            type=type, **kwargs
         )
 
     def __call__(self, parser, namespace, values, option_string=None):
         if self.nargs == 0 and self.const is not None:
             setattr(namespace, self.dest, self.const)
         elif self.nargs == 1:
-            setattr(namespace, self.dest, util.coerce_from_iter(values))
+            setattr(namespace, self.dest, util.from_iter(values))
         else:
             setattr(namespace, self.dest, values)
-        # set flag to indicate user has set this argument
-        setattr(namespace, self.dest+self.flag_suffix, False)
+        # set additional flag to indicate user has set this argument
+        setattr(namespace, self.dest+self.default_value_suffix, False)
 
+class PathAction(RecordDefaultsAction):
+    """:py:class:`~argparse.Action` that performs shell environment variable 
+    expansion and resolution of relative paths, using 
+    :func:`src.util.filesystem.resolve_path`.
+    """
+    call_on_defaults = True
 
-class CLIHandler(object):
-    def __init__(self, code_root, cli_config, partial_defaults=None):
-        self.code_root = code_root
-        self.config = dict()
-        self.partial_defaults = partial_defaults
-        self.parser_groups = dict()
-        # no way to get this from public interface? _actions of group
-        # contains all actions for entire parser
-        self.parser_args_from_group = collections.defaultdict(list)
-        # manually track args requiring custom postprocessing (even if default
-        # is used, so can't do with action=.. in argument)
-        self.custom_types = collections.defaultdict(list)
-        if isinstance(cli_config, six.string_types):
-            # we were given a path to config file, instead of file's contents
-            if not os.path.isabs(cli_config):
-                cli_config = os.path.join(code_root, cli_config)
-            cli_config = util.read_json(cli_config)
-        self.parser = self.make_parser(cli_config)
-
-    def iter_cli_actions(self):
-        for arg_gp in self.parser_args_from_group:
-            for arg in self.parser_args_from_group[arg_gp]:
-                yield arg
-
-    def iteritems_cli(self, group_nm=None):
-        if not group_nm:
-            _groups = self.parser_groups
-        else:
-            _groups = util.coerce_to_iter(group_nm)
-        for group in _groups:
-            for action in self.parser_args_from_group[group]:
-                key = action.dest
-                yield (key, self.config[key])
+    def __call__(self, parser, namespace, values, option_string=None):
+        # config = CLIConfigManager()
+        path = util.from_iter(values)
+        if path is None:
+            path = ''
+        # resolve paths in PathManager init, after env vars set
```

Comment:
Remove commented-out lines?

Reply:
Done in NOAA-GFDL/MDTF-diagnostics@7b47104f5077872e6b20e1af63fb487da281db9b